### PR TITLE
Add endpoint to get all articles

### PR DESCRIPTION
**Pull request by jasonbarrows**
In addition to closing ticket 5, this PR addresses fixes some file name typos and a refactor of two route not found endpoint tests into a single 404 test for non-existent paths.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -57,11 +62,35 @@ describe('GET /api/topics', () => {
...
+      .expect(200)
+      .then((response) => {
+        const { articles } = response.body
+
+        expect(articles).toBeInstanceOf(Array);
+        expect(articles).toHaveLength(13);
+        expect(articles).toBeSortedBy('created_at', { descending: true });
+
+        articles.forEach((article) => {
+          expect(article.body).not.toBeDefined();
```
Comment (lewispricey) [inline __tests__/app.test.js:80]: Great test, asserting that the body isn't being returned!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -57,11 +62,35 @@ describe('GET /api/topics', () => {
...
-      .get('/api/bananas')
-      .expect(404);
+      .get('/api/articles')
+      .expect(200)
+      .then((response) => {
+        const { articles } = response.body
+
+        expect(articles).toBeInstanceOf(Array);
+        expect(articles).toHaveLength(13);
+        expect(articles).toBeSortedBy('created_at', { descending: true });
```
Comment (lewispricey) [inline __tests__/app.test.js:77]: Awesome use of the jest sorted package here to check the default ordering of responses.

File: models/articles.models.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const db = require('../db/connection');
+
+exports.selectAllArticles = () => {
+  return db.query(
+    `SELECT a.article_id, a.title, a.topic, a.author, a.created_at, a.votes, a.article_img_url, CAST (COUNT (c.comment_id) AS INTEGER) AS comment_count
```
Comment (lewispricey) [inline models/articles.models.js:5]: Nice work on this query, casting the result as a number is particularly tricky 🙌

File: __tests__/app.test.js [hunk 4]
```diff
@@ -7,6 +7,17 @@ const app = require('../app');
 beforeEach(() => seed(data));
 afterAll(() => db.end());
 
+describe('ALL non-existent paths', () => {
```
Comment (lewispricey) [inline __tests__/app.test.js:10]: Nice to see this broken out to its own test block since it doesn't directly relate to any of the other endpoints

**Review by lewispricey — APPROVED** [review]
Awesome work on this ticket Jason. Your tests are flexible yet super thorough which is awesome to see and the code is super easy to read.

Great work on the DB query (this ones pretty tricky especialy to return a number) and I like the use of giving articles the alias of a.

Good to merge when you're ready 🚀

**Comment by jasonbarrows** [comment]
Thanks for all your comments Lewis.  I appreciate the valuable feedback.

**Pull request merged**